src/config.rs

```diff
@@ -92,20 +99,21 @@ impl PbftConfig {
 pub fn load_pbft_config(block_id: BlockId, service: &mut Service) -> PbftConfig {
     let mut config = PbftConfig::default();
 
-    let settings: HashMap<String, String> = service
-        .get_settings(
-            block_id,
+    let settings: HashMap<String, String> = retry_until_ok(config.exponential_retry_base, || {
```

Comment:
It would be good to add a log statement before you start the retry loop so you can see if we get stuck on a step. Same for other places you use retry_until_ok